src/main/java/io/antmedia/rest/RestServiceBase.java

```diff
@@ -438,50 +438,45 @@ private static void removeEmptyPlayListItems(Broadcast broadcast)
 	 * @return
 	 */
 	protected Result updateStreamSource(String streamId, Broadcast broadcast) {
+		logger.debug("Updating camera info for stream {}", broadcast.getStreamId());
 
-		boolean result = false;
-
-		boolean resultStopStreaming = false;
-
-		logger.debug("update cam info for stream {}", broadcast.getStreamId());
-
-		if(checkStreamUrl(broadcast.getStreamUrl()))
-		{
-			Broadcast broadcastInDB = getDataStore().get(streamId);
-			if (broadcastInDB != null)
-			{
-				resultStopStreaming = checkStopStreaming(broadcastInDB);
-
-				waitStopStreaming(broadcastInDB, resultStopStreaming);
-
-				if (AntMediaApplicationAdapter.IP_CAMERA.equals(broadcast.getType())) {
-					String rtspURL = connectToCamera(broadcast).getMessage();
-
-					if (rtspURL != null) {
+		if (!checkStreamUrl(broadcast.getStreamUrl())) {
+			return new Result(false);
+		}
 
-						String authparam = broadcast.getUsername() + ":" + broadcast.getPassword() + "@";
-						String rtspURLWithAuth = RTSP + authparam + rtspURL.substring(RTSP.length());
-						logger.info("new RTSP URL: {}" , rtspURLWithAuth);
-						broadcast.setStreamUrl(rtspURLWithAuth);
-					}
-				}
+		Broadcast broadcastInDB = getDataStore().get(streamId);
+		if (broadcastInDB == null) {
+			streamId = streamId.replaceAll("[\n|\r|\t]", "_");
+			logger.info("Broadcast with stream id: {} is null", streamId);
+			return new Result(false);
```

Comment:
Let's add message to the Result object because developers can save hours by just reading the message